### PR TITLE
Implement twig hooks into cart page and widget

### PR DESCRIPTION
**Pull request by kulczy**
- temporary i moved cart button and offcanvas renderer into shared folder - i think it should be change into live component

### CODE REVIEW

**Review by TheMilek — APPROVED** [review]

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/cart/index.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            remove:
+                template: '@SyliusShop/cart/page/body/details/table_head/remove.html.twig'
+            item:
+                template: '@SyliusShop/cart/page/body/details/table_head/item.html.twig'
+            unit_price:
+                template: '@SyliusShop/cart/page/body/details/table_head/unit_price.html.twig'
+            quantity:
+                template: '@SyliusShop/cart/page/body/details/table_head/quantity.html.twig'
+            total:
+                template: '@SyliusShop/cart/page/body/details/table_head/total.html.twig'
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/cart/index.yaml]:
```suggestion
        'sylius_shop.cart.page.body.details.table':
            head:
                template: '@SyliusShop/cart/page/body/details/table.head.html.twig'

        'sylius_shop.cart.page.body.details.table.head':
            remove:
                template: '@SyliusShop/cart/page/body/details/table/head/remove.html.twig'
            item:
                template: '@SyliusShop/cart/page/body/details/table/head/item.html.twig'
            unit_price:
                template: '@SyliusShop/cart/page/body/details/table/head/unit_price.html.twig'
            quantity:
                template: '@SyliusShop/cart/page/body/details/table/head/quantity.html.twig'
            total:
                template: '@SyliusShop/cart/page/body/details/table/head/total.html.twig'
```
the same for `body`

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details.html.twig [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+{{ form_start(form, {'action': path('sylius_shop_cart_save'), 'attr': {'novalidate': 'novalidate', 'id': form.vars.id}}) }}
+{{ form_errors(form) }}
+<input type="hidden" name="_method" value="PATCH"/>
+{{ form_row(form._token) }}
+{{ form_end(form, {'render_rest': false}) }}
+
+<div class="col-12 col-xl-8 mb-4">
+    {% hook 'details' %}
+</div>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details.html.twig]: Shouldn't this be within the form?

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/table_body/unit_price.html.twig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{% import "@SyliusShop/shared/macro/money.html.twig" as money %}
+
+{% set item = hookable_metadata.context.item %}
+{% set original_price_to_display = sylius_order_item_original_price_to_display(item) %}
+
+<td class="text-black-50 text-end">
+    {% if original_price_to_display is not null %}
+        <span {{ sylius_test_html_attribute('cart-product-regular-unit-price') }}>
+                <span>{{ money.convertAndFormat(original_price_to_display) }}</span>
+            </span>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/table_body/unit_price.html.twig]:
```suggestion
        <span {{ sylius_test_html_attribute('cart-product-regular-unit-price') }}>
            {{ money.convertAndFormat(original_price_to_display) }}
        </span>
```
Why is the span doubled?

File: src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/cart_button.html.twig [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{% import "@SyliusShop/shared/macro/money.html.twig" as money %}
+{% from '@SyliusShop/shared/icon.html.twig' import icon as icon %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/shared/layout/base/header/content/cart_button.html.twig]:
```suggestion
{% from '@SyliusShop/shared/icon.html.twig' import icon %}
```

File: src/Sylius/Bundle/ShopBundle/templates/cart/widget/body/items.html.twig [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{% set cart = hookable_metadata.context.cart %}
+
+{% if cart.empty %}
+    {{ 'sylius.ui.your_cart_is_empty'|trans }}.
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/widget/body/items.html.twig:4]:
```suggestion
    {{ 'sylius.ui.your_cart_is_empty'|trans }}
```

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/footer/suggestions.html.twig [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<div>
+    <div class="h2 mb-5">{{ 'sylius.ui.you_may_also_like'|trans }}</div>
+    {{ render(url('sylius_shop_partial_product_index_latest', {'count': 4, 'template': '@SyliusShop/Product/horizontal_list.html.twig'})) }}
+</div>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/footer/suggestions.html.twig]: IMO this begs for a twig component. Not in this PR though.

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/body/sidebar/summary.html.twig [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{% import "@SyliusShop/shared/macro/money.html.twig" as money %}
+
+{% set cart = hookable_metadata.context.cart %}
+
+{% set itemsSubtotal = cart.getItemsSubtotal() %}
+{% set taxIncluded = cart.getTaxIncludedTotal() %}
+{% set taxExcluded = cart.getTaxExcludedTotal() %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/body/sidebar/summary.html.twig]:
```suggestion
```
Unused

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/coupon.html.twig [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{% if form.promotionCoupon is defined %}
+    <div class="mb-4">
+        <div class="p-4 bg-light">
+            <div class="input-group" {{ sylius_test_html_attribute('cart-promotion-coupon') }}>
+                {{ form_widget(form.promotionCoupon, sylius_test_form_attribute('cart-promotion-coupon-input')|sylius_merge_recursive({'attr': {'form': main_form, 'placeholder': 'sylius.ui.enter_your_code'|trans~'...'}})) }}
+                <button class="btn btn-outline-gray" type="submit" {{ sylius_test_html_attribute('apply-coupon-button') }} form="{{ main_form }}">{{ 'sylius.ui.apply_coupon'|trans }}</button>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/coupon.html.twig]: Where is the `main_form` coming from?

File: src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/coupon.html.twig [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{% if form.promotionCoupon is defined %}
+    <div class="mb-4">
+        <div class="p-4 bg-light">
+            <div class="input-group" {{ sylius_test_html_attribute('cart-promotion-coupon') }}>
+                {{ form_widget(form.promotionCoupon, sylius_test_form_attribute('cart-promotion-coupon-input')|sylius_merge_recursive({'attr': {'form': main_form, 'placeholder': 'sylius.ui.enter_your_code'|trans~'...'}})) }}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/cart/page/body/details/coupon.html.twig]:
```suggestion
                {{ form_widget(form.promotionCoupon, sylius_test_form_attribute('cart-promotion-coupon-input')|sylius_merge_recursive({'attr': {'form': main_form, 'placeholder': 'sylius.ui.enter_your_code'|trans ~ '...'}})) }}
```

**Review by Wojdylak — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @kulczy!

**Pull request merged**